hooks/MockReturnTypeUpdater.php

```diff
@@ -51,7 +51,16 @@ public static function afterMethodCallAnalysis(
                 && $first_arg->right instanceof PhpParser\Node\Scalar\String_
                 && $first_arg->right->value[0] === '['
             ) {
-                $fq_class_name = $first_arg->left->class->getAttribute('resolvedName');
+                /** @var PhpParser\Node\Expr\ClassConstFetch $left */
+                $left = $first_arg->left;
+                $fq_class_name = $left->class->getAttribute('resolvedName');
+            } elseif ($first_arg instanceof PhpParser\Node\Scalar\String_) {
```

Comment:
This is the addition I made for #1 which allows, and interprets the class properly when these syntaxes are used:

`Mockery::mock('ClassName[method1,method2]');`
`Mockery::mock('ClassName');`